GraphicsView/include/CGAL/Qt/qglviewer_impl.h

```diff
@@ -205,33 +204,35 @@ This method is automatically called once, before the first call to paintGL().
 
 Overload init() instead of this method to modify viewer specific OpenGL state.
 
-If a 4.3 context could not be set, a 2.1 context will be used instead. 
+If a 4.3 context could not be set, a ES 2.0 context will be used instead. 
  \see `isOpenGL_4_3()`
 */
 CGAL_INLINE_FUNCTION
 void CGAL::QGLViewer::initializeGL() {
-  QSurfaceFormat format;
-  format.setDepthBufferSize(24);
-  format.setStencilBufferSize(8);
-  format.setVersion(4,3);
-  format.setProfile(QSurfaceFormat::CompatibilityProfile);
-  format.setSamples(0);
-  format.setOption(QSurfaceFormat::DebugContext);
-  context()->setFormat(format);
-  bool created = context()->create();
-  if(!created || context()->format().profile() != QSurfaceFormat::CompatibilityProfile) {
-    // impossible to get a 4.3 compatibility profile, retry with 2.0
-    format.setVersion(2,1);
-    context()->setFormat(format);
-    created = context()->create();
+  QSurfaceFormat format = context()->format();
+  context()->format().setOption(QSurfaceFormat::DebugContext);
+  if ( !context()->isValid()
+    || format.majorVersion() != 4
+    || QCoreApplication::arguments().contains(QStringLiteral("--old")))
+
+  {
+    format.setDepthBufferSize(24);
+    format.setStencilBufferSize(8);
+    format.setVersion(2,0);
+    format.setRenderableType(QSurfaceFormat::OpenGLES);
+    format.setSamples(0);
+    format.setOption(QSurfaceFormat::DebugContext);
+    QSurfaceFormat::setDefaultFormat(format);
+    needNewContext();
```

Comment:
@maxGimeno I think you should `qDebug()` something, here, to be able to know which mode is used by the app.